addons/website/migrations/12.0.1.0/pre-migration.py

```diff
@@ -1,11 +1,194 @@
+    # Create a column to remember where do new views come from
+    col_name = openupgrade.get_legacy_name("bs4_migrated_from")
+    table_name = env["ir.ui.view"]._table
+    if not openupgrade.column_exists(env.cr, table_name, col_name):
+        openupgrade.logged_query(
+            env.cr,
+            "ALTER TABLE %s ADD COLUMN %s INTEGER",
+            (AsIs(table_name), AsIs(col_name)),
+        )
+    # Find report views, which should never be converted here
+    report_names = env["ir.actions.report"].search([
+        ("report_name", "!=", False),
+        ("report_type", "=like", "qweb-%"),
+    ]).mapped("report_name")
+    # Find updatable views, to be excluded; standard addon update is enough
+    udpatable_ids = env["ir.model.data"].search([
+        ("model", "=", "ir.ui.view"),
+        ("noupdate", "=", False),
+    ]).mapped("res_id")
+    # In Odoo v11, a page can be related to multiple websites, and the related
+    # view doesn't need to be related to a website (although it could be);
+    # in v12, a page is related to a view, which is related (or not) to a sigle
+    # website. Usually it would be impossible to link a page to multiple
+    # websites using UI in v11, so it's not like a
+    env.cr.execute(
+        """SELECT r.website_id, p.view_id
+           FROM website_website_page_rel AS r
+           INNER JOIN website_page AS p ON r.website_page_id = p.id
+           INNER JOIN ir_ui_view AS v ON p.view_id = v.id
+           WHERE v.website_id IS NULL
+           ORDER BY r.website_id DESC""",
+    )
+    for page_website_id, view_id in env.cr.fetchall():
+        # If a page happens to be in different websites, only the first one
+        # will be stored in the view; the others will be lost, which isn't a
+        # problem since v11 didn't support multi-websites
+        openupgrade.logged_query(
+            env.cr,
+            "UPDATE ir_ui_view SET website_id = %s WHERE id = %s",
+            (page_website_id, view_id),
+        )
+    # Find views to convert
+    all_views = env['ir.ui.view'].search([
+        ("active", "=", True),
+        ("id", "not in", udpatable_ids),
+        ("key", "not in", report_names),
+        ("type", "=", "qweb"),
+    ])
+    # Find available websites
+    env.cr.execute("SELECT id FROM website")
+    website_ids = list(chain.from_iterable(env.cr.fetchall()))
+    # Convert in place views that already have a website
+    env.cr.execute(
+        "SELECT id FROM %s WHERE id IN %s AND website_id IS NOT NULL",
+        (AsIs(table_name), tuple(all_views.ids)),
+    )
+    views_with_website = env["ir.ui.view"].browse(
+        list(chain.from_iterable(env.cr.fetchall())),
+        prefetch=all_views._prefetch,
+    )
+    for view in views_with_website:
+        new_arch = convert_string_bootstrap_3to4(view.arch_base)
+        new_arch = _apply_v11_dummy_image_replacement(
+            image_replacements,
+            new_arch,
+        )
+        view.arch_base = new_arch
```

Comment:
Why not using `arch_db` field directly for avoiding any kind on manipulation on the content?

Reply:
Actually I wanted to let Odoo do its stuff, whatever it is. It seems to handle updates in several translations automatically. :man_shrugging:  Well, I didn't want to dig so much, just use the one that is exposed to the user and seems to have a proper inverse method.

Reply:
You should change `arch_db` here. This will trigger the compute `_compute_arch` that will calculate the `arch`, and then this will trigger the compute `_compute_arch_base` that will calculate the `arch_base` and so on.

Reply:
That seems to be the same as what the inverse method of `arch_base` does, so I see no difference. Anyways I'll change it, you're the pros.